src/cmdstan/arguments/arg_num_threads.hpp

```diff
@@ -10,8 +10,8 @@ class arg_num_threads : public int_argument {
   arg_num_threads() : int_argument() {
     _name = "num_threads";
     _description = std::string("Number of threads available to the program.");
-    _default = "-1";
-    _default_value = -1;
+    _default = "1";
+    _default_value = 1;
```

Comment:
Can we encode here a minimal and maximal value of 1 whenever `STAN_THREADS` is absent (I am not familiar with the argument processing stuff too much; so this is a naive q)? Whenever `STAN_THREADS` is defined, then there is no upper limit. I recall that others were not happy with using -1 for "use all cores". Instead "0" was preferred, but we probably keep that consistent.